cscs-checks/prgenv/helloworld.py

```diff
@@ -89,7 +90,12 @@ def num_ranks(match):
 
     @rfm.run_before('compile')
     def setflags(self):
-        envname = self.current_environ.name.replace('-nompi', '')
+        if '-nompi-nocuda' in self.current_environ.name:
+            envname = self.current_environ.name.replace('-nompi-nocuda', '')
+        elif '-nocuda' in self.current_environ.name:
+            envname = self.current_environ.name.replace('-nocuda', '')
+        else:
+            envname = self.current_environ.name.replace('-nompi', '')
```

Comment:
Replace all this with the following one liner:

```suggestion
        envname = re.sub(r'(PrgEnv-\w+).*', lambda m: m.group(1), self.current_environ.name)
```